api/hardwaremanagement/v1alpha1/node_pools.go

```diff
@@ -62,7 +62,8 @@ type Properties struct {
 
 // GenerationStatus represents the observed generation for an operator.
 type GenerationStatus struct {
-	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+	ObservedGeneration int64  `json:"observedGeneration,omitempty"`
+	SpecHash           string `json:"specHash,omitempty"`
```

Comment:
Why do we need both a generation and a hash?  Doesn't the generation change every time an update is applied?

Reply:
The SpecHash and ObservedGeneration each serve a different purpose. The SpecHash field captures the hash of the spec changes triggered by the hardware template and enables the operator to quickly detect when updates to the NodePool are required. It also differentiates the changes made by the plugin.  The ObservedGeneration field allows each operator to confirm that it has processed the most recent version of the spec. This is essential when multiple operators are involved, as each can independently confirm it has reconciled the latest changes. The ObservedGeneration is updated after a successful spec update.

Reply:
Based on my discussion with Don, the plugin will not update the NodePool spec, and I will modify this PR accordingly.

Reply:
PR is updated.